data/core/init.lua

```diff
@@ -1166,24 +785,20 @@ function core.pop_clip_rect()
   renderer.set_clip_rect(x, y, w, h)
 end
 
-
-function core.normalize_to_project_dir(filename)
-  filename = common.normalize_path(filename)
-  if common.path_belongs_to(filename, core.project_dir) then
-    filename = common.relative_path(core.project_dir, filename)
-  end
-  return filename
-end
-
+-- legacy interface
+function core.root_project() return core.projects[1] end
+function core.normalize_to_project_dir(path) return core.root_project():normalize_path(path) end
+function core.project_absolute_path(path) return core.root_project() and core.root_project():absolute_path(path) or system.absolute_path(path) end
```

Comment:
I wonder if we should just avoid these, so that people think if they actually want the root project or a more appropriate one.

Should we add a function that tries to return the correct project for a file path?

Reply:
These are here as legacy functions so we don't have to go in any fix a ton of stuff. I can add a deprecation warning, if you like.

And sure; would probably be reasonable to get a project with a path. Done.